hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesClient.java

```diff
@@ -608,4 +670,79 @@ public String toString() {
             return String.format("%s:%s", ip, port);
         }
     }
+
+    final class StsMonitor implements Runnable {
+        private String latestResourceVersion;
+        private RuntimeContext latestRuntimeContext;
+
+        @Override
+        public void run() {
+            String stsUrlString = String.format("%s/apis/apps/v1/namespaces/%s/statefulsets", kubernetesMaster,
+                    namespace);
+            JsonObject jsonObject = callGet(stsUrlString);
+            latestResourceVersion = jsonObject.get("metadata").asObject().getString("resourceVersion",
+                    null);
+            latestRuntimeContext = extractStsList(jsonObject);
+            LOGGER.info("Initializing cluster topology tracker with initial context: "
+                    + latestRuntimeContext);
+            clusterTopologyIntentTracker.update(UNKNOWN,
+                    latestRuntimeContext.getDesiredNumberOfMembers(),
+                    latestRuntimeContext.getReadyReplicas());
+            while (true) {
```

Comment:
This is not a busy-spin loop, so it won't burn CPU cycles. It will block until some input is available in line 706
```
while ((message = watchResponse.nextLine()) != null) {
```
`nextLine()` blocks until a whole line is read from the k8s API server (or `IOException`/EOF occurs)